### ISSUE TITLE
module 'json' not found while running the emulator

### ISSUE THREAD

**Issue by lwiechec** [koreader/koreader#7409]
* KOReader version: latest from git (sha 4118234)
* Device: emulator

#### Issue

running emulator fails with the following message:
```
[luke:~/src/koreader] master 127 ± ./kodev run -s=kobo-aura-one
[*] Building KOReader…
************ CMAKE_MAKE_PROGRAM: "make" **********
make -C base
make[1]: Entering directory '/home/luke/src/koreader/base'
************ CMAKE_MAKE_PROGRAM: "make" **********
************ Building for MACHINE: "x86_64-linux-gnu-debug" **********
************ PATH: "/home/luke/.sdkman/candidates/groovy/current/bin:/home/luke/bin:/home/luke/work/tools/apache-maven/bin:/home/luke/work/tools/apache-ant/bin:/home/luke/work/jdk/default_8/bin:/home/luke/.local/bin:/home/luke/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/luke/.dotnet/tools:/home/luke/.rvm/bin:/home/luke/go/bin/:/home/luke/.fzf/bin" **********
************ CHOST: "" **********
# set up some needed paths and links
install -d build/x86_64-linux-gnu-debug/{cache,history,clipboard,fonts} /home/luke/src/koreader/base/build/x86_64-linux-gnu-debug/plugins/evernote.koplugin/thrift
ln -sf /home/luke/src/koreader/base/thirdparty/kpvcrlib/cr3.css build/x86_64-linux-gnu-debug/data/
# setup Evernote SDK
cd /home/luke/src/koreader/base/thirdparty/evernote-sdk-lua/build/x86_64-linux-gnu-debug/evernote-sdk-lua-prefix/src/evernote-sdk-lua && \
	cp -r *.lua evernote /home/luke/src/koreader/base/build/x86_64-linux-gnu-debug/plugins/evernote.koplugin && \
	cp thrift/*.lua /home/luke/src/koreader/base/build/x86_64-linux-gnu-debug/plugins/evernote.koplugin/thrift
test -e build/x86_64-linux-gnu-debug/rocks/share/lua/5.1/re.lua && chmod 664 build/x86_64-linux-gnu-debug/rocks/share/lua/5.1/re.lua || true  # hot fix re.lua permission
make[1]: Leaving directory '/home/luke/src/koreader/base'
install -d koreader-emulator-x86_64-linux-gnu-debug/koreader
rm -f koreader-emulator-x86_64-linux-gnu-debug/koreader/git-rev; echo "v2021.02-62-g41182349_2021-03-10" > koreader-emulator-x86_64-linux-gnu-debug/koreader/git-rev
cp -f base/ev_replay.py koreader-emulator-x86_64-linux-gnu-debug/koreader/
[*] create symlink instead of copying files in development mode
cd koreader-emulator-x86_64-linux-gnu-debug/koreader && \
	bash -O extglob -c "ln -sf ../../base/build/x86_64-linux-gnu-debug/!(cache) ."
[*] install front spec only for the emulator
cd koreader-emulator-x86_64-linux-gnu-debug/koreader/spec && test -e front || \
	ln -sf ../../../../spec ./front
cd koreader-emulator-x86_64-linux-gnu-debug/koreader/spec/front/unit && test -e data || \
	ln -sf ../../test ./data
for f in reader.lua setupkoenv.lua frontend resources defaults.lua datastorage.lua l10n tools README.md COPYING; do \
	ln -sf ../../$f koreader-emulator-x86_64-linux-gnu-debug/koreader/; \
done
[*] Install plugins
cp -r plugins/. koreader-emulator-x86_64-linux-gnu-debug/koreader/plugins/.
for d in $(ls koreader-emulator-x86_64-linux-gnu-debug/koreader/plugins); do \
	test -d plugins/$d || rm -rf koreader-emulator-x86_64-linux-gnu-debug/koreader/plugins/$d ; done
[*] Install resources
cp -r -pL resources/fonts/. koreader-emulator-x86_64-linux-gnu-debug/koreader/fonts/.
install -d koreader-emulator-x86_64-linux-gnu-debug/koreader/{screenshots,data/{dict,tessdata},fonts/host,ota}
[*] Running KOReader with arguments:  ...
~/src/koreader/koreader-emulator-x86_64-linux-gnu-debug/koreader ~/src/koreader
---------------------------------------------
                launching...
  _  _____  ____                _
 | |/ / _ \|  _ \ ___  __ _  __| | ___ _ __
 | ' / | | | |_) / _ \/ _` |/ _` |/ _ \ '__|
 | . \ |_| |  _ <  __/ (_| | (_| |  __/ |
 |_|\_\___/|_| \_\___|\__,_|\__,_|\___|_|

 It's a scroll... It's a codex... It's KOReader!

 [*] Current time: 03/13/21-14:45:11
 [*] Version: v2021.02-62-g41182349_2021-03-10

ffi.load: blitbuffer
ffi.load: SDL2
ffi.load: SDL2
03/13/21-14:45:11 INFO  platform thirdparty apps
-> Goldendict (goldendict), role: dict, available: false

Starting SDL in /home/luke/src/koreader/base/build/x86_64-linux-gnu-debug/
SDL: no gamecontrollers connected
03/13/21-14:45:11 DEBUG refresh on physical rectangle 0 0 1404 1872
ffi.load: libs/liblodepng.so
03/13/21-14:45:11 INFO  initializing for device Emulator
03/13/21-14:45:11 INFO  framebuffer resolution: {
    ["w"] = 1404,
    ["h"] = 1872,
}
ffi.load: libs/libmupdf.so
ffi.load: libs/libwrap-mupdf.so
ffi.load: libs/libfreetype.so.6
ffi.load: libs/libharfbuzz.so.0
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  Noto Sans
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  Nimbus Roman No9 L
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  Nimbus Mono
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  Noto Sans Arabic UI
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  FreeSans
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  Nimbus Sans L
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  Noto Naskh Arabic
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  Noto Serif
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  Dingbats
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  Droid Sans Mono
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  FreeSerif
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  Noto Sans Devanagari UI
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  Noto Sans CJK SC
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  Standard Symbols L
03/13/21-14:45:11 DEBUG FONTNAMES ADD:  NotoSans Nerd Font
03/13/21-14:45:11 DEBUG Found font: NotoSans-Regular.ttf in ./fonts/noto/NotoSans-Regular.ttf
03/13/21-14:45:11 DEBUG Found font: NotoSans-Regular.ttf in ./fonts/noto/NotoSans-Regular.ttf
03/13/21-14:45:11 DEBUG Found font: NotoSansCJKsc-Regular.otf in ./fonts/noto/NotoSansCJKsc-Regular.otf
03/13/21-14:45:11 DEBUG ImageWidget: _render'ing resources/icons/mdlight/notice-info.svg 94 94
03/13/21-14:45:11 DEBUG renderSVG resources/icons/mdlight/notice-info.svg 1.9583333333333 48 48 > 94 94 0 0
03/13/21-14:45:11 DEBUG cache image|resources/icons/mdlight/notice-info.svg|94|94
03/13/21-14:45:11 DEBUG ImageWidget: initial offsets 0 0
03/13/21-14:45:11 DEBUG show widget: table: 0x7f1ce5410560
03/13/21-14:45:11 DEBUG setDirty nil from widget table: 0x7f1ce5410560 w/ NO region 
03/13/21-14:45:11 DEBUG setDirty via a func from widget table: 0x7f1ce5410560
03/13/21-14:45:11 DEBUG Found font: NotoSans-Bold.ttf in ./fonts/noto/NotoSans-Bold.ttf
03/13/21-14:45:11 DEBUG Found font: NotoSans-Regular.ttf in ./fonts/noto/NotoSans-Regular.ttf
03/13/21-14:45:11 DEBUG Found font: NotoSans-Bold.ttf in ./fonts/noto/NotoSans-Bold.ttf
03/13/21-14:45:11 DEBUG Found font: NotoSans-Regular.ttf in ./fonts/noto/NotoSans-Regular.ttf
03/13/21-14:45:11 DEBUG Found font: DroidSansMono.ttf in ./fonts/droid/DroidSansMono.ttf
./luajit: frontend/apps/reader/modules/readerdictionary.lua:10: module 'json' not found:
	no field package.preload['json']
	no file 'common/json.lua'
	no file 'rocks/share/lua/5.1/json.lua'
	no file 'frontend/json.lua'
	no file './json.lua'
	no file '/usr/local/share/luajit-2.1.0-beta3/json.lua'
	no file '/usr/local/share/lua/5.1/json.lua'
	no file '/usr/local/share/lua/5.1/json/init.lua'
	no file 'common/json.so'
	no file 'common/json.dll'
	no file '/usr/lib/lua/json.so'
	no file 'rocks/lib/lua/5.1/json.so'
	no file './json.so'
	no file '/usr/local/lib/lua/5.1/json.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
	[C]: in function 'require'
	frontend/apps/reader/modules/readerdictionary.lua:10: in main chunk
	[C]: in function 'require'
	frontend/apps/reader/readerui.lua:31: in main chunk
	[C]: in function 'require'
	./reader.lua:288: in main chunk
	[C]: at 0x5560cce81a64
```

#### Notes

Hello, this is my first foray into the KOReader - I have never used/built it before. I try to follow the instructions on how to build.

I guess that fetching of the third-party libraries is not going well.

I am building on Ubuntu (I have installed all the deps).

##### `crash.log` (if applicable)
N/A

**Comment by Frenzie** [comment]
Which version of Ubuntu?

**Comment by NiLuJe** [comment]
That's not even rapidjson failing to build, so I smell some LuaRocks shenanigans ;).

**Comment by lwiechec** [comment]
> Which version of Ubuntu?

It's 18.04.5 LTS

**Comment by lwiechec** [comment]
> That's not even rapidjson failing to build, so I smell some LuaRocks shenanigans ;).

ha, shenaningans sound - intimidating :) I wonder how can I proceed?

**Comment by NiLuJe** [comment]
Double-checking that you're actually running a supported LuaRocks version and that it's set up properly would be a good start.

**Comment by Frenzie** [comment]
Debian 9/Ubuntu 16.04 has 2.2.x, Debian 10/Ubuntu 18.04 up to today have 2.4.x. That shouldn't be an issue.

**Comment by NiLuJe** [comment]
I was specicially thinging about the "properly set up" part (.e.g, Lua 5.1/LuaJIT) ^^.

**Comment by Frenzie** [comment]
Let me rephrase. If it wasn't installed prior to `sudo apt install luarocks` then it's most likely set up right. ^_^

**Comment by lwiechec** [comment]
If it helps: here are the Lua-related DEB packages installed on the build machine:

```
[luke:~/src/koreader]↥ master ± aptitude search lua | grep '^i'
i  liblua5.1-0 - Shared library for the Lua interpreter version 5.1
i A liblua5.1-0-dev - Development files for the Lua language version 5.1
i  liblua5.2-0 - Shared library for the Lua interpreter version 5.2
i  liblua5.3-0 - Shared library for the Lua interpreter version 5.3
i  libluajit-5.1-2 - Just in time compiler for Lua - library version
i  libluajit-5.1-common - Just in time compiler for Lua - common files
i  libtexlua52 - TeX Live: Lua 5.2, modified for use with LuaTeX
i  libtexluajit2 - TeX Live: LuaJIT, modified for use with LuaJITTeX
i A lua-any - helper script for shebang lines in Lua scripts
i  lua-json - JSON decoder/encoder for Lua
i A lua-lpeg - LPeg library for the Lua language
i A lua-sec - SSL socket library for the Lua language
i A lua-socket - TCP/UDP socket library for the Lua language
i  lua5.2 - Simple, extensible, embeddable programming language
i  lua5.3 - Simple, extensible, embeddable programming language
i  luarocks - deployment and management system for Lua modules
```

**Comment by Frenzie** [comment]
Have you tried also installing lua5.1?

**Comment by lwiechec** [comment]
thanks for the tip. 
After installing `lua-5.1` and re-cloning the build of the installer succeeded.

**Comment by Frenzie** [comment]
By default luarocks depends on Lua 5.1, but it'll also be happy with 5.2 or 5.3. It should probably be added to the apt-get list in the building README.
